src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -941,7 +941,7 @@ CHIP_ERROR ConnectivityManagerImpl::ProvisionWiFiNetwork(const char * ssid, cons
                         ChipLogDetail(DeviceLayer, "Got IP address on interface: %s IP: %s", ifName,
                                       event.InternetConnectivityChange.address);
 
-                        PlatformMgr().PostEvent(&event);
+                        ReturnErrorOnFailure(PlatformMgr().PostEvent(&event));
```

Comment:
This is a slightly odd thing to do in a function with an `exit:` cleanup block.  I guess we know `err` is null here?  Still seems a bit odd.

In particular, it's not clear to me why skipping the dhclient bits if this PostEvent fails is better than just setting `ret` to a failure here and moving forward.....  especially given the comment after those dhclient bits, which claims this function should returns succcess even if some bits inside it fail.